### PR TITLE
make HUD bubble/shield button optional

### PR DESCRIPTION
**Pull request by SilverfishVR**
Adds an menu item in "Settings" to toggle the HUD shield/bubble button on or off.
Shield button remains available in the tablet or toolbar, it is called "HUD shield button" but suggestions are welcome

fixes https://github.com/vircadia/vircadia/issues/1210 but makes it optional.

### CODE REVIEW

File: scripts/system/bubble.js [hunk 1]
```diff
@@ -46,6 +46,27 @@
...
+        menuItemName: menuItemName,
+        isCheckable: true,
+        isChecked: AvatarInputs.showBubbleTools
+    });
+    Menu.menuItemEvent.connect(onToggleHudShieldButton);
+    AvatarInputs.showBubbleToolsChanged.connect(showBubbleToolsChanged);
+
+    function onToggleHudShieldButton(menuItem) {
+        if (menuItem == menuItemName) {
+            AvatarInputs.setShowBubbleTools(Menu.isOptionChecked(menuItem))
```
Comment (ctrlaltdavid) [inline scripts/system/bubble.js:62]:
```suggestion
            AvatarInputs.setShowBubbleTools(Menu.isOptionChecked(menuItem));
```

File: scripts/system/bubble.js [hunk 2]
```diff
@@ -46,6 +46,27 @@
...
+        menuName: "Settings",
+        menuItemName: menuItemName,
+        isCheckable: true,
+        isChecked: AvatarInputs.showBubbleTools
+    });
+    Menu.menuItemEvent.connect(onToggleHudShieldButton);
+    AvatarInputs.showBubbleToolsChanged.connect(showBubbleToolsChanged);
+
+    function onToggleHudShieldButton(menuItem) {
+        if (menuItem == menuItemName) {
```
Comment (ctrlaltdavid) [inline scripts/system/bubble.js:61]:
```suggestion
        if (menuItem === menuItemName) {
```

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]
Just a couple of minor items.

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-Interface-PR1314-f821989-_a7dee4bb.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-f821989-.tar_a7dee4bb.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-f821989-_a7dee4bb.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-f821989-.tar_a7dee4bb.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-f821989-_a7dee4bb.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-f821989-_a7dee4bb.exe

File: scripts/system/bubble.js [hunk 3]
```diff
@@ -46,6 +46,27 @@
         });
     }
 
+    //create a menu item in "Setings" to toggle the bubble/shield HUD button 
+    var menuItemName = "HUD shield button";
```
Comment (digisomni) [inline scripts/system/bubble.js:50]:
```suggestion
    var menuItemName = "HUD Shield Button";
```

**Comment by digisomni** [comment]
Works fine, just that one change and I think it'll be good to go.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-49b9154-_4d21cedf.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-49b9154-.tar_4d21cedf.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-49b9154-.tar_4d21cedf.Z

**build (macOS-latest, client)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-Interface-PR1314-49b9154-_4d21cedf.dmg

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-49b9154-_4d21cedf.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/vircadia_vircadia/PR1314/Vircadia-PR1314-49b9154-_4d21cedf.exe

**Review by digisomni — APPROVED** [review]

**Pull request merged**